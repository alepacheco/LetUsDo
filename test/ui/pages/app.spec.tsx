/* eslint-disable no-undef, import/no-unresolved */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MyApp from 'src/../pages/_app';

jest.mock('src/utils/analytics', () => () => {});

describe('<MyApp />', () => {
  const FakeComponent: any = ({ foo }: any) => <p>{foo}</p>;
  const fakeRouter: any = {};

  it('renders', () => {
    const wrapper = shallow(
      <MyApp Component={FakeComponent} pageProps={{ foo: 'bar' }} router={fakeRouter} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(toJson(wrapper.find(FakeComponent).dive())).toMatchSnapshot();
  });
});
