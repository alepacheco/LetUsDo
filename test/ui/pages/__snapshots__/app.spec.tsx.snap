// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyApp /> renders 1`] = `
<Fragment>
  <Head>
    <link
      href="https://m.stripe.com"
      rel="preconnect"
    />
    <script
      async={true}
      src="https://js.stripe.com/v3/"
    />
    <link
      crossOrigin="anonymous"
      href="https://maxcdn.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
      integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T"
      rel="stylesheet"
    />
    <link
      href="/static/favicon.ico"
      rel="icon"
      type="image/x-icon"
    />
    <link
      href="/static/favicon.ico"
      rel="shortcut icon"
      type="image/png"
    />
    <title>
      Let Us Do
    </title>
  </Head>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <FakeComponent
      foo="bar"
    />
  </Provider>
</Fragment>
`;

exports[`<MyApp /> renders 2`] = `
<p>
  bar
</p>
`;
